packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -491,12 +614,62 @@ class MPChartData {
   /// Datapoint entries for each external memory value.
   final List<Entry> externalHeap = <Entry>[];
 
+  /// Prune entries plotted based on number of sample
+  /// minutes to display.
+  void prune(int minutesToDisplay) {
+    assert(minutesToDisplay > 0);
+
+    final len = used.length;
+
+    // Entries should match.
+    // All entries lengths are the same.
+    assert(len == capacity.length && len == externalHeap.length);
+    // All entries starting timestamps match,
+    assert(used.isNotEmpty &&
+        used[0].x == capacity[0].x &&
+        capacity[0].x == externalHeap[0].x);
+    // All entries ending timestamps match.
+    assert(used.isNotEmpty &&
+        used[len - 1].x == capacity[len - 1].x &&
+        capacity[len - 1].x == externalHeap[len - 1].x);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = used.last.x;
+    var index = len - 1;
+    for (; index > 0; index--) {
+      final sample = used[index];
+      final timestamp = sample.x;
+
+      if ((timeLastSample - timestamp) > minutesToDisplay) break;
+    }
+
+    if (index != 0) {
+      used.removeRange(0, index);
+      capacity.removeRange(0, index);
+      externalHeap.removeRange(0, index);
+    }
+  }
+
   /// Add each entry to its corresponding trace.
   void addTraceEntries({
     Entry capacityValue,
     Entry usedValue,
     Entry externalValue,
+    int minutesToDisplay,
   }) {
+    if (!_pruning &&
+        externalHeap.isNotEmpty &&
+        (externalValue.x - externalHeap.first.x) > minutesToDisplay) {
+      _pruning = true;
+//      assert(externalValue.x - externalHeap.last.x <= minutesToDisplay);
```

Comment:
can this be removed?

Reply:
Uncommented the assert its a good to check validity data displayed in the chart visible view is correct.